### PR TITLE
add replyTo prop for MessagingInput

### PR DESCRIPTION
**Pull request by jonjonray**
**Jira:**
[M5G-148](https://clever.atlassian.net/browse/M5G-148?atlOrigin=eyJpIjoiZDIwMWY0OGE2M2Q3NGIyMTkzMTY5M2Q0YmM1MTMxZWMiLCJwIjoiaiJ9)

**Overview:**
Adding replyTo prop for MessagingInput to support announcement replies for milestone 1a. 

**Screenshots/GIFs:**

Screen Shot 2021-04-06 at 7 01 26 PM

Screen Shot 2021-04-07 at 12 33 10 PM

Screen Shot 2021-04-07 at 12 32 46 PM

Original design
Screen Shot 2021-04-06 at 1 08 15 PM

**Testing:**
- Tested changes with voiceover, input + label are still being read correctly.
- Old messaging input remains unchanged

- [x] Unit tests
- Manual tests:
  - [x] Chrome
  - [x] Safari
  - [ ] IE11

**Roll Out:**
100%
- Before merging:
  - [x] Updated docs
  - [x] Bumped version in `package.json`
    - Breaking change?
      - If it is a beta component run `npm version minor`
      - If the component is not in beta run `npm version major`
    - New component or backward-compatible component feature change? Run `npm version minor`
   
- After merging:
  - [ ] Deployed updated docs (`make deploy-docs`)
  - [ ] Posted in #eng if I made a breaking change to a beta component

### CODE REVIEW

File: src/MessagingInput/MessagingInput.tsx [hunk 1]
```diff
@@ -60,36 +63,43 @@ const MessagingInputRenderFunction: React.ForwardRefRenderFunction<MessagingInpu
 
   return (
     <FlexBox className={cx(cssClass("Container"), className)} column alignItems={ItemAlign.START}>
-      <label htmlFor={TEXT_FIELD_NAME} className={cssClass("TextFieldLabel")}>
+      <label
+        htmlFor={TEXT_FIELD_NAME}
+        className={cssClass(replyTo ? "TextFieldLabelHidden" : "TextFieldLabel")}
```
Comment (jonjonray) [inline src/MessagingInput/MessagingInput.tsx:68]: using class for hiding so the label is still read by a screenreader if using a placeholder instead

File: src/MessagingInput/MessagingInput.tsx [hunk 2]
```diff
@@ -60,36 +63,43 @@ const MessagingInputRenderFunction: React.ForwardRefRenderFunction<MessagingInpu
...
-            }
-          }}
-          onBlur={onBlur}
-          autoResize
-          // The field starts with `rows + 1` rows, so
-          //  passing in 0 gets us the desired starting height.
-          rows={0}
-        />
+            }}
+            placeholder={replyTo ? labelText : ""}
```
Comment (jonjonray) [inline src/MessagingInput/MessagingInput.tsx:95]: We may want to rethink this if we ever want to support a placeholder prop for this input but works for our existing use-case

File: src/MessagingInput/MessagingInput.less [hunk 3]
```diff
@@ -9,13 +9,39 @@
...
+  border-bottom: none;
+  .padding--xs();
+}
+
 .MessagingInput--TextFieldLabel {
   color: @neutral_medium_gray;
   .margin--bottom--2xs();
 }
 
+.MessagingInput--TextFieldLabelHidden {
```
Comment (jonjonray) [inline src/MessagingInput/MessagingInput.less:30]: Using this to hide the label so screen readers will stick pick up the label content

File: src/MessagingInput/MessagingInput.less [hunk 4]
```diff
@@ -9,13 +9,73 @@
...
+  &:hover,
+  &:focus,
+  &:active {
+    background-color: @neutral_gray;
+    color: @neutral_off_white;
+  }
+}
+
+.MessagingInput--Reply--CloseIcon {
+  // covers possible range of FA sizes
```
Comment (nbhatia823) [inline src/MessagingInput/MessagingInput.less:49]: 🎉

File: src/MessagingInput/MessagingInput.less [hunk 5]
```diff
@@ -9,13 +9,73 @@
...
+  margin: -1px;
+  overflow: hidden;
+  clip: rect(0, 0, 0, 0);
+  white-space: nowrap; /* added line */
+  border: 0;
+  /* stylelint-enable */
+}
+
 .MessagingInput--TextField {
-  .margin--right--xs();
```
Comment (nbhatia823) [inline src/MessagingInput/MessagingInput.less:18]: What was the reason for dropping this?

File: docs/components/MessagingInputView.jsx [hunk 6]
```diff
@@ -141,6 +157,12 @@ export default class MessagingInputView extends React.PureComponent {
             description: "If true, manually disables the send button",
             optional: true,
           },
+          {
+            name: "replyTo",
+            type: "React.ReactNode",
+            description: "Optional prop to use for message reply content",
+            optional: true,
+          },
```
Comment (nbhatia823) [inline docs/components/MessagingInputView.jsx:165]: Think you had written this part before you added the `onReplyCancel` prop

File: src/MessagingInput/MessagingInput.less (hunk 3 above)
Comment (nbhatia823) [reply src/MessagingInput/MessagingInput.less:64]: Very nice!

**Review by nbhatia823 — APPROVED** [review]
Great! Looks like we were on the same page with how to implement - this looks awesome. And I appreciate the clarity/conciseness of your code as well as the thoughtfulness towards accessibility. One comment about adding in the new prop to the docs

File: docs/components/MessagingInputView.jsx (hunk 6 above)
Comment (jonjonray) [reply docs/components/MessagingInputView.jsx:165]: oh shoot good catch

**Comment by jonjonray** [comment]
Great callouts. Fixing some small formatting issues for 1 & 2!

**Comment by jonjonray** [comment]
Added extra screenshots with max height and mobile revisions

**Pull request merged**